airbyte-integrations/connectors/source-freshdesk/source_freshdesk/streams.py

```diff
@@ -0,0 +1,273 @@
+        if "updated_at" in stream_state:
+            params[self.state_filter] = stream_state["updated_at"]
+        else:
+            params[self.state_filter] = self.start_date
+        return params
+
+
+class Agents(FreshdeskStream):
+
+    def path(self, **kwargs) -> str:
+        return "agents"
+
+
+class Companies(FreshdeskStream):
+
+    def path(self, **kwargs) -> str:
+        return "companies"
+
+
+class Contacts(IncrementalFreshdeskStream):
+    state_filter = "_updated_since"
+
+    def path(self, **kwargs) -> str:
+        return "contacts"
+
+
+class Groups(FreshdeskStream):
+
+    def path(self, **kwargs) -> str:
+        return "groups"
+
+
+class Roles(FreshdeskStream):
+
+    def path(self, **kwargs) -> str:
+        return "roles"
+
+
+class Skills(FreshdeskStream):
+
+    def path(self, **kwargs) -> str:
+        return "skills"
+
+
+class TimeEntries(FreshdeskStream):
+
+    def path(self, **kwargs) -> str:
+        return "time_entries"
+
+
+class Tickets(IncrementalFreshdeskStream):
+    ticket_paginate_limit = 300
+
+    def path(self, **kwargs) -> str:
+        return "tickets"
+    
+    def request_params(
+        self, stream_state: Mapping[str, Any], stream_slice: Mapping[str, any] = None, next_page_token: Mapping[str, Any] = None
+    ) -> MutableMapping[str, Any]:
+        params = super().request_params(stream_state=stream_state, stream_slice=stream_slice, next_page_token=next_page_token)
+        params.update({
+            "order_type": "asc",  # ASC order, to get the old records first
+            "order_by": "updated_at",
+        })
+        if next_page_token and "updated_since" in next_page_token:
+            params["updated_since"] = next_page_token["updated_since"]
+        return params
+    
+    def parse_link_params(self, link_query_params: Mapping[str, List[str]]) -> Mapping[str, Any]:
+        params = super().parse_link_params(link_query_params)
+        if "updated_since" in link_query_params:
+            params["updated_since"] = link_query_params["updated_since"][0]
+        return params
+    
+    def read_records(
+        self,
+        sync_mode: SyncMode,
+        cursor_field: List[str] = None,
+        stream_slice: Mapping[str, Any] = None,
+        stream_state: Mapping[str, Any] = None,
+    ) -> Iterable[Mapping[str, Any]]:
+        """
+        Read ticket records
+
+        This block extends Incremental stream to overcome '300 page' server error.
+        Since the Ticket endpoint has a 300 page pagination limit, after 300 pages, update the parameters with
+        query using 'updated_since' = last_record, if there is more data remaining.
+        """
+        stream_state = stream_state or {}
+        pagination_complete = False
+
+        next_page_token = None
+        with AirbyteSentry.start_transaction("read_records", self.name), AirbyteSentry.start_transaction_span("read_records"):
```

Comment:
Would you mind explaining why copying large amounts of the parent stream`read_records()`is needed instead of invoking `super().read_records()`

Reply:
You made me re-think how I migrated this particular stream and I ended up overriding the `next_page_token` method instead of the `read_records` one. In the end, I had to intervene the `next_page_token` if the next page to get was above 300. The method now has a comment that was ported from the original implementation. See [b0bcf0e](https://github.com/airbytehq/airbyte/pull/12334/commits/b0bcf0e00310eb439c80526604439211cd0adf97).